generative/smrd/models/models.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch.nn as nn
+from mutils import get_sigmas
+from .layers import *
+from .normalization import get_normalization
+
+
+class UNet(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.logit_transform = config.data.logit_transform
+        self.rescaled = config.data.rescaled
+        self.norm = get_normalization(config, conditional=False)
+        self.ngf = ngf = config.model.ngf
+        self.num_classes = config.model.num_classes
+        self.act = act = get_act(config)
+        self.register_buffer("sigmas", get_sigmas(config))
+        self.config = config
+
+        self.begin_conv = nn.Conv2d(config.data.channels, ngf, 3, stride=1, padding=1)
+        self.normalizer = self.norm(ngf, self.num_classes)
+
+        self.end_conv = nn.Conv2d(ngf, config.data.channels, 3, stride=1, padding=1)
+
+        self.res1 = nn.ModuleList(
```

Comment:
You could use `Sequential` instead of `ModuleList` here and elsewhere in this class, then you wouldn't need `_compute_cond_module`.